download_weather_streaming.py

```
@@ -0,0 +1,308 @@
+        try:
+            # Get sensors for this location
+            sensors = await datasource.get_sensors(location, parameters=param_types)
+            
+            # Process data in monthly chunks
+            current_start = start_date or datetime.now(timezone.utc) - timedelta(days=30)
+            final_end = end_date or datetime.now(timezone.utc)
+            
+            while current_start <= final_end:
+                chunk_end = min(
+                    current_start.replace(day=1) + timedelta(days=32),
+                    final_end
```

Ruff (0.12.2)

Trailing comma missing

Add trailing comma

(COM812)